Tools/ci/generate_board_targets_json.py

```diff
@@ -25,15 +25,15 @@
                     help='Pretty output instead of a single line')
 parser.add_argument('-g', '--groups', dest='group', action='store_true',
                     help='Groups targets')
-parser.add_argument('-f', '--filter', dest='filter', help='comma separated list of board names to use instead of all')
+parser.add_argument('-f', '--filter', dest='filter', help='comma separated list of board or target names to use instead of all')
 
 args = parser.parse_args()
 verbose = args.verbose
 
-board_filter = []
+target_filter = []
 if args.filter:
-    for board in args.filter.split(','):
-        board_filter.append(board)
+    for target in args.filter.split(','):
+        target_filter.append(target)
```

Comment:
As an alternative, check if the target has less than two underscores, and if it has less, add a underscore tot he target string. This way you add the underscore here and not need rely on the user inputing this in the filter